starter-code/app.js

```diff
@@ -1,79 +1,73 @@
-var express = require('express');
-var path = require('path');
-var favicon = require('serve-favicon');
-var logger = require('morgan');
-var cookieParser = require('cookie-parser');
-var bodyParser = require('body-parser');
-var app = express();
-
-var index = require('./routes/index');
-var users = require('./routes/users');
-const passportRouter = require("./routes/passportRouter");
-//mongoose configuration
-const mongoose = require("mongoose");
-mongoose.connect("mongodb://localhost/passport-local");
-//require the user model
-const User = require("./models/user");
-const session       = require("express-session");
-const bcrypt        = require("bcrypt");
-const passport      = require("passport");
-const LocalStrategy = require("passport-local").Strategy;
-const flash = require("connect-flash");
-
-
-
-
-
-//enable sessions here
-
-
-
-
-//initialize passport and session here
-
-
+const express = require('express');
+const path = require('path');
+const favicon = require('serve-favicon');
+const logger = require('morgan');
+const cookieParser = require('cookie-parser');
+const bodyParser = require('body-parser');
+const app = express();
+const configurePassport = require('./misc/passport');
+
+const index = require('./routes/index');
+const users = require('./routes/users');
+const passportRouter = require('./routes/passportRouter');
+// mongoose configuration
+const mongoose = require('mongoose');
+mongoose.connect('mongodb://localhost/passport-local');
+// require the user model
+
+const session = require('express-session');
+
+const passport = require('passport');
+
+const flash = require('connect-flash');
+const authRoutes = require('./routes/authRoutes');
+
+// enable sessions here
+
+mongoose.Promise = Promise;
+mongoose.connect('mongodb://localhost/passport-local', {
+  keepAlive: true,
+  reconnectTries: Number.MAX_VALUE
+});
 
+// initialize passport and session here
 
+app.use(session({
+  secret: 'our-passport-local-strategy-app',
+  resave: true,
+  saveUninitialized: true
+}));
```

Comment:
The session should be after the cookie parser. Else it will create new session everytime you are doing a request.